pkg/proxy/session.go

```diff
@@ -162,6 +167,8 @@ func (s *Session) loopReader(tasks *RequestChan, d *Router) (err error) {
 		s.incrOpTotal()
 
 		if tasks.Buffered() > maxPipelineLen {
+			s.incrFails()
+			s.flushOpStats(true)
```

Comment:
The function flushOpStats is not thread-safe (routine-safe).
It can only be called by the writer, I mean this line is not necessary and can be removed.

Reply:
thanks. I ignored that the writer will flush op stats when it exits. 
i have updated it.